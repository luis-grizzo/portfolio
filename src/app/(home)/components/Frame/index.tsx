'use client'

import Image from 'next/image'
import { motion } from 'framer-motion'

import { useColor } from '@/hooks/useColor'

import { verticalHidden, verticalVisible } from '@/utils/animations'

export const Frame = () => {
  const { color } = useColor()

  return (
    <motion.div
      initial={verticalHidden}
      animate={verticalVisible}
      exit={verticalHidden}
      className={`flex items-end justify-center relative w-full h-72 xsm:h-96 lg:h-full rounded-3xl overflow-hidden before:content-[''] before:bg-${color}-600 before:h-[65%] before:w-full before:rounded-3xl`}
    >
      <motion.div
        initial={{ y: '100%', opacity: 0 }}
        animate={{ y: '2%', opacity: 1 }}
        exit={{ y: '100%', opacity: 0 }}
        transition={{ type: 'spring', bounce: 0.2 }}
        className="absolute h-full w-max"
      >
        <Image
          priority
          width={2943}
          height={2022}
          className="h-full w-max"
          src="/portrait.png"
          alt="LuÃ­s Grizzo's portrait"
        />
      </motion.div>
    </motion.div>
  )
}
