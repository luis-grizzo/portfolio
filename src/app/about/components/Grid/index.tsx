'use client'

import { cloneElement, useState, useEffect, useRef } from 'react'
import { AnimatePresence, motion } from 'framer-motion'

import { useColor } from '@/hooks/useColor'

import { verticalHidden, verticalVisible } from '@/utils/animations'
import { techs } from '@/utils/techs'

export const Grid = (): React.ReactElement => {
  const gridRef = useRef<HTMLDivElement>(null)

  const { color } = useColor()

  const [isFullLeftScroll, setIsFullLeftScroll] = useState(false)
  const [isFullRightScroll, setIsFullRightScroll] = useState(false)

  const handleScroll = (): void => {
    const offsetWidth = gridRef.current?.offsetWidth ?? 0
    const scrollLeft = gridRef.current?.scrollLeft ?? 0
    const scrollWidth = gridRef.current?.scrollWidth ?? 0

    scrollLeft === 0 ? setIsFullLeftScroll(true) : setIsFullLeftScroll(false)

    offsetWidth + scrollLeft >= scrollWidth
      ? setIsFullRightScroll(true)
      : setIsFullRightScroll(false)
  }

  useEffect(() => {
    handleScroll()

    gridRef.current?.addEventListener('scroll', handleScroll)

    return () => gridRef.current?.removeEventListener('scroll', handleScroll)
  }, [])

  return (
    <div className="flex relative h-full w-full overflow-hidden">
      <div
        ref={gridRef}
        className="grid grid-cols-[repeat(12,_auto)] lg:grid-cols-[1fr_1fr_1fr] auto-rows-[1fr] gap-6 w-full h-full overflow-y-hidden"
      >
        {techs.map((tech, index) => (
          <motion.div
            key={tech.name}
            initial={verticalHidden}
            animate={verticalVisible}
            exit={verticalHidden}
            transition={{ delay: index * 0.1 }}
            className={`flex items-center justify-center w-40 aspect-square lg:aspect-auto lg:w-full rounded-[1.25rem] text-contrast_color_lightTheme dark:text-contrast_color_darkTheme bg-shape_color_lightTheme dark:bg-shape_color_darkTheme hover:bg-${color}-600 transition-colors`}
          >
            {cloneElement(tech.icon, {
              size: 60,
              className: 'gi__icon'
            })}
          </motion.div>
        ))}
      </div>

      <AnimatePresence>
        {!isFullLeftScroll && (
          <motion.div
            key="left"
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="absolute top-0 left-0 h-full w-1/6 bg-gradient-to-r from-background_color_lightTheme dark:from-background_color_darkTheme"
          />
        )}

        {!isFullRightScroll && (
          <motion.div
            key="right"
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="absolute top-0 right-0 h-full w-1/6 bg-gradient-to-l from-background_color_lightTheme dark:from-background_color_darkTheme"
          />
        )}
      </AnimatePresence>
    </div>
  )
}
