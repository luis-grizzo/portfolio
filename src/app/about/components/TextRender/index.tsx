'use client'

import { useState, useEffect, useRef } from 'react'
import { AnimatePresence, motion } from 'framer-motion'

import { horizontalHidden, horizontalVisible } from '@/utils/animations'

interface TextRenderProps {
  texts: TextProps[]
}

interface TextProps {
  id: number
  paragraph: React.ReactElement
}

export const TextRender = ({ texts }: TextRenderProps) => {
  const textContainerRef = useRef<HTMLDivElement>(null)

  const [isFullScrolled, setIsFullScrolled] = useState(false)
  const [isTopScroll, setIsTopScroll] = useState(false)

  const handleScroll = (): void => {
    const offsetHeight = textContainerRef.current?.offsetHeight ?? 0
    const scrollTop = textContainerRef.current?.scrollTop ?? 0
    const scrollHeight = textContainerRef.current?.scrollHeight ?? 0

    scrollTop === 0 ? setIsTopScroll(true) : setIsTopScroll(false)

    offsetHeight + scrollTop >= scrollHeight
      ? setIsFullScrolled(true)
      : setIsFullScrolled(false)
  }

  useEffect(() => {
    handleScroll()

    textContainerRef.current?.addEventListener('scroll', handleScroll)

    return () =>
      textContainerRef.current?.removeEventListener('scroll', handleScroll)
  }, [])

  return (
    <div className="flex relative h-[40%] overflow-hidden">
      <div
        ref={textContainerRef}
        className="flex flex-col items-start gap-6 overflow-auto overflow-x-hidden h-full w-full pr-2 text-xl text-text_color_lightTheme dark:text-text_color_darkTheme custom-scroll"
      >
        {texts.map((text, index) => (
          <motion.div
            key={text.id}
            initial={horizontalHidden}
            animate={horizontalVisible}
            exit={horizontalHidden}
            transition={{ delay: (index + 1) * 0.1 }}
          >
            {text.paragraph}
          </motion.div>
        ))}
      </div>

      <AnimatePresence mode="wait">
        {!isTopScroll && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="absolute top-0 left-0 h-1/4 w-[calc(100%_-_0.5rem)] bg-gradient-to-b from-background_color_lightTheme to-transparent dark:from-background_color_darkTheme"
          />
        )}
      </AnimatePresence>

      <AnimatePresence mode="wait">
        {!isFullScrolled && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="absolute bottom-0 left-0 h-1/4 w-[calc(100%_-_0.5rem)] bg-gradient-to-t from-background_color_lightTheme to-transparent dark:from-background_color_darkTheme"
          />
        )}
      </AnimatePresence>
    </div>
  )
}
