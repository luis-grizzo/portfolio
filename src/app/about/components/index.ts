export * from './Grid'
export * from './TextRender'
