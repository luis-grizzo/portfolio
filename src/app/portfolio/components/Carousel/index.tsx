'use client'

import { cloneElement, useState, useEffect } from 'react'
import Image from 'next/image'
import { AnimatePresence, motion } from 'framer-motion'
import { FaCss3 } from 'react-icons/fa6'
import { BiLogoTypescript } from 'react-icons/bi'
import { DiJavascript1 } from 'react-icons/di'
import { MdStar, MdCode } from 'react-icons/md'

import { useColor } from '@/hooks/useColor'
import { useResize } from '@/hooks/useResize'

import { verticalHidden, verticalVisible } from '@/utils/animations'

import { FetchReposProps } from '@/types'

interface CarouselProps {
  items: FetchReposProps[]
}

export const Carousel = ({ items }: CarouselProps) => {
  const { color } = useColor()
  const { isDesktop } = useResize()

  const [activeItem, setActiveItem] = useState(0)

  const handleLanguageIcon = (language: string) => {
    const selectedLanguage = {
      css: <FaCss3 />,
      typescript: <BiLogoTypescript />,
      javascript: <DiJavascript1 />
    }[language.toLowerCase()]

    return cloneElement(selectedLanguage || <MdCode />, {
      size: 16,
      className: 'text-shape_color_lightTheme dark:text-shape_color_darkTheme'
    })
  }

  useEffect(() => {
    const timer = setTimeout(() => {
      setActiveItem((prevState) => {
        if (prevState + 1 === items.length) return 0
        else return prevState + 1
      })
    }, 5000)

    return () => clearTimeout(timer)
  }, [activeItem])

  return (
    <div className="grid grid-cols-[auto,_1fr] gap-6 relative h-full w-full overflow-hidden pb-5 before:content-[''] before:absolute before:bottom-0 before:left-0 before:w-full before:h-5 before:z-10 before:bg-gradient-to-t from-background_color_lightTheme to-transparent dark:from-background_color_darkTheme">
      <div className="flex flex-col justify-center gap-6 z-10">
        {items.map((item, index) => (
          <motion.button
            key={item.id}
            initial={verticalHidden}
            animate={verticalVisible}
            transition={{ delay: index * 0.1 }}
            className={`w-5 h-5 rounded-[50%] transition-colors ${
              index === activeItem
                ? `bg-${color}-600 pointer-events-none`
                : `bg-contrast_color_lightTheme dark:bg-contrast_color_darkTheme hover:bg-${color}-600`
            }`}
            onClick={() => setActiveItem(index)}
          />
        ))}
      </div>

      <div className="flex items-center justify-center relative h-96 lg:h-full w-full">
        <AnimatePresence mode="wait">
          {items.map(
            (item, index) =>
              index === activeItem && (
                <motion.div
                  key={item.id}
                  initial={verticalHidden}
                  animate={verticalVisible}
                  exit={verticalHidden}
                  className={`flex flex-col justify-between items-start absolute text-left h-full w-full p-12 rounded-3xl bg-shape_color_lightTheme dark:bg-shape_color_darkTheme z-0 ${
                    index !== activeItem && 'top-[100%]'
                  }`}
                >
                  <div className="self-end flex items-center gap-2">
                    <Image
                      src={item.owner.avatar_url}
                      width={500}
                      height={500}
                      alt={`Avatar de ${item.owner.login}`}
                      className="w-12 rounded-[50%]"
                    />

                    <span className="font-bold text-contrast_color_lightTheme dark:text-contrast_color_darkTheme">
                      {item.owner.login}
                    </span>
                  </div>

                  <div className="flex flex-col gap-4">
                    <div className="flex flex-col gap-2">
                      <h2 className="text-4xl font-bold text-contrast_color_lightTheme dark:text-contrast_color_darkTheme">
                        {item.name}
                      </h2>

                      <div className="flex flex-wrap items-center gap-2">
                        <div className="flex items-center gap-1 px-1.5 rounded-[0.375rem] bg-contrast_color_lightTheme dark:bg-contrast_color_darkTheme">
                          <MdStar
                            size={16}
                            className="text-shape_color_lightTheme dark:text-shape_color_darkTheme"
                          />

                          <span className="text-shape_color_lightTheme dark:text-shape_color_darkTheme text-base">
                            {item.stargazers_count}
                          </span>
                        </div>

                        {item.language && (
                          <div className="flex items-center gap-1 px-1.5 rounded-[0.375rem] bg-contrast_color_lightTheme dark:bg-contrast_color_darkTheme">
                            {handleLanguageIcon(item.language)}

                            <span className="text-shape_color_lightTheme dark:text-shape_color_darkTheme text-base">
                              {item.language}
                            </span>
                          </div>
                        )}
                      </div>

                      <div className="flex items-center gap-2">
                        <a
                          href={item.html_url}
                          target="_blank"
                          className={`text-${color}-600 font-bold transition-colors hover:text-${color}-900`}
                        >
                          Repository
                        </a>

                        {item.homepage && (
                          <>
                            <span className="w-2 h-2 rounded-[50%] bg-contrast_color_lightTheme dark:bg-contrast_color_darkTheme" />

                            <a
                              href={item.homepage}
                              target="_blank"
                              className={`text-${color}-600 font-bold transition-colors hover:text-${color}-900`}
                            >
                              Homepage
                            </a>
                          </>
                        )}
                      </div>
                    </div>

                    <p className="text-contrast_color_lightTheme dark:text-contrast_color_darkTheme">
                      {item.description}
                    </p>
                  </div>
                </motion.div>
              )
          )}
        </AnimatePresence>
      </div>
    </div>
  )
}
