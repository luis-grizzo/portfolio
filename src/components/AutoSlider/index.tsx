'use client'

import { useEffect, useRef, useState } from 'react'
import { AnimatePresence, motion } from 'framer-motion'

import { timeUnits } from '@/constants/timeUnits'

import { defaultTransition } from '@/utils/animations'

interface AutoSliderProps {
  children: React.ReactNode
}

type FullScrollTypes = 'left' | 'right' | null

export function AutoSlider({ children }: AutoSliderProps) {
  const containerRef = useRef<HTMLDivElement>(null)
  const contentRef = useRef<HTMLDivElement>(null)

  const [fullScrolled, setFullScrolled] = useState<FullScrollTypes>('left')

  function handleScroll() {
    const { current: container } = containerRef
    const { current: content } = contentRef

    if (container && content) {
      let value!: FullScrollTypes

      if (container.scrollLeft === 0) value = 'left'
      else if (
        container.scrollLeft ===
        content.clientWidth - container.clientWidth
      )
        value = 'right'
      else value = null

      setFullScrolled(value)
    }
  }

  function handleAnimation() {
    const { current: container } = containerRef
    const { current: content } = contentRef

    if (container && content) {
      let destination!: number

      if (container.scrollLeft === 0)
        destination = content.clientWidth - container.clientWidth
      else destination = 0

      container.scrollTo({
        left: destination,
        behavior: 'smooth'
      })
    }
  }

  useEffect(() => {
    const { current: container } = containerRef

    if (container) container.addEventListener('scroll', handleScroll)

    const timer = setInterval(handleAnimation, timeUnits.second * 3)

    return () => {
      if (container) container.removeEventListener('scroll', handleScroll)

      clearInterval(timer)
    }
  }, [])

  return (
    <div className="relative w-full">
      <AnimatePresence mode="popLayout">
        {fullScrolled !== 'left' && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            transition={defaultTransition}
            key="left"
            data-testid="left-fade"
            className="absolute left-0 top-0 w-4 h-full bg-gradient-to-r from-neutral-950 z-50"
          />
        )}

        {fullScrolled !== 'right' && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            transition={defaultTransition}
            key="right"
            data-testid="right-fade"
            className="absolute right-0 top-0 w-4 h-full bg-gradient-to-l from-neutral-950 z-50"
          />
        )}
      </AnimatePresence>

      <div ref={containerRef} role="tabpanel" className="overflow-hidden -z-50">
        <div
          ref={contentRef}
          role="tablist"
          className="flex items-center gap-4 w-min"
        >
          {children}
        </div>
      </div>
    </div>
  )
}
