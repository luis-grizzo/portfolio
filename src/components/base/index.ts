export * from './Card'
export * from './LinkButton'
export * from './Logo'
