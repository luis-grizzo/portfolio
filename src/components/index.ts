export * from './AutoSlider'
export * from './Button'
export * from './Card'
export * from './Footer'
export * from './Footnote'
export * from './Hero'
export * from './LinkButton'
export * from './LittleSunshine'
export * from './Navbar'
