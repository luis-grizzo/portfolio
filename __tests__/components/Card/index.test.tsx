import { render, screen } from '@testing-library/react'

import { Card, CardProps } from '@/components'

jest.useFakeTimers()

describe('Card', () => {
  beforeAll(() => {
    const fakeDate = new Date('2024-07-06T18:25:47Z')

    jest.setSystemTime(fakeDate)
  })

  afterAll(() => {
    jest.clearAllTimers()
  })

  const mockProject: CardProps['project'] = {
    forks_count: 10,
    language: 'language',
    watchers_count: 10,
    created_at: '2024-05-21T14:42:05Z',
    description: 'description',
    fork: false,
    homepage: 'homepage',
    html_url: 'html_url',
    license: { name: 'license name' },
    name: 'name',
    pushed_at: '2024-07-03T18:25:47Z',
    stargazers_count: 10,
    topics: ['topic']
  }

  it('should match snapshot', () => {
    const { container } = render(<Card project={mockProject} />)

    expect(container).toMatchSnapshot()
  })

  it('should not render the new project badge if createdAt data is null', () => {
    const newMockProject: CardProps['project'] = {
      ...mockProject,
      created_at: null
    }

    render(<Card project={newMockProject} />)

    const eNewProjectBadge = screen.queryByText(/new project/i)

    expect(eNewProjectBadge).not.toBeInTheDocument()
  })

  it('should not render the recently pushed badge if pushedAt data is null', () => {
    const newMockProject: CardProps['project'] = {
      ...mockProject,
      pushed_at: null
    }

    render(<Card project={newMockProject} />)

    const eRecentlyPushedBadge = screen.queryByText(/recently pushed/i)

    expect(eRecentlyPushedBadge).not.toBeInTheDocument()
  })
})
