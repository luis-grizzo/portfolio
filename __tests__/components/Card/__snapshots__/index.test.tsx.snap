// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-4 py-8 first:border-t-1 lg:card-top-border border-b-1 border-neutral-400/10"
  >
    <div
      class="flex flex-wrap items-center gap-x-4 gap-y-2"
    >
      <span
        class="flex items-center gap-2 w-max text-sm italic text-purple-400"
      >
        <svg
          fill="currentColor"
          height="14"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.716 1.329a1.341 1.341 0 0 1 2.109 1.55L15.147 9h4.161c1.623 0 2.372 2.016 1.143 3.075L8.102 22.721a1.148 1.148 0 0 1-1.81-1.317L8.996 15H4.674c-1.619 0-2.37-2.008-1.148-3.07l12.19-10.6Zm.452 1.595L4.51 13.061a.25.25 0 0 0 .164.439h5.45a.749.749 0 0 1 .692 1.041l-2.559 6.066 11.215-9.668a.25.25 0 0 0-.164-.439H14a.75.75 0 0 1-.687-1.05Z"
          />
        </svg>
        New project
      </span>
      <span
        class="flex items-center gap-2 text-sm italic text-neutral-400"
      >
        <svg
          fill="currentColor"
          height="14"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.75 2.75V4.5h1.975c.351 0 .694.106.984.303l1.697 1.154c.041.028.09.043.14.043h4.102a.75.75 0 0 1 0 1.5H20.07l3.366 7.68a.749.749 0 0 1-.23.896c-.1.074-.203.143-.31.206a6.296 6.296 0 0 1-.79.399 7.349 7.349 0 0 1-2.856.569 7.343 7.343 0 0 1-2.855-.568 6.205 6.205 0 0 1-.79-.4 3.205 3.205 0 0 1-.307-.202l-.005-.004a.749.749 0 0 1-.23-.896l3.368-7.68h-.886c-.351 0-.694-.106-.984-.303l-1.697-1.154a.246.246 0 0 0-.14-.043H12.75v14.5h4.487a.75.75 0 0 1 0 1.5H6.763a.75.75 0 0 1 0-1.5h4.487V6H9.275a.249.249 0 0 0-.14.043L7.439 7.197c-.29.197-.633.303-.984.303h-.886l3.368 7.68a.75.75 0 0 1-.209.878c-.08.065-.16.126-.31.223a6.077 6.077 0 0 1-.792.433 6.924 6.924 0 0 1-2.876.62 6.913 6.913 0 0 1-2.876-.62 6.077 6.077 0 0 1-.792-.433 3.483 3.483 0 0 1-.309-.221.762.762 0 0 1-.21-.88L3.93 7.5H2.353a.75.75 0 0 1 0-1.5h4.102c.05 0 .099-.015.141-.043l1.695-1.154c.29-.198.634-.303.985-.303h1.974V2.75a.75.75 0 0 1 1.5 0ZM2.193 15.198a5.414 5.414 0 0 0 2.557.635 5.414 5.414 0 0 0 2.557-.635L4.75 9.368Zm14.51-.024c.082.04.174.083.275.126.53.223 1.305.45 2.272.45a5.847 5.847 0 0 0 2.547-.576L19.25 9.367Z"
          />
        </svg>
        license name
      </span>
    </div>
    <div
      class="flex items-center justify-between gap-2"
    >
      <h2
        class="w-full text-2xl truncate"
      >
        name
      </h2>
      <svg
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        />
      </svg>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <a
        class="underline underline-offset-2 text-sm text-blue-400 hover:text-blue-200 transition-colors"
        href="html_url"
        target="_blank"
      >
        Repository
      </a>
      <a
        class="underline underline-offset-2 text-sm text-blue-400 hover:text-blue-200 transition-colors"
        href="homepage"
        target="_blank"
      >
        Homepage
      </a>
    </div>
    <p
      class="text-base text-pretty text-neutral-400"
    >
      description
    </p>
    <div
      class="flex flex-wrap items-center gap-2"
    >
      <span
        class="flex items-center px-2 py-1 text-xs bg-neutral-800/60 text-neutral-50 border-1 border-neutral-400/10 rounded-lg"
      >
        topic
      </span>
    </div>
  </div>
</div>
`;
